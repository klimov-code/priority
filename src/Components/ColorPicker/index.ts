export { ColorPicker } from './ColorPicker';
export { Container, Circle } from './styled';
