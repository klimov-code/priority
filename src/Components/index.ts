export { Priority } from './Priority';
export { Editor } from './Editor';
export { Viewer } from './Viewer';
export { Header } from './Header';
