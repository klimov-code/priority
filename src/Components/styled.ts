import styled from 'styled-components';

const Text = styled.span`
  display: block;
  font-size: 13px;
  line-height: 18px;
  color: #454c54;
`;

const TextBold = styled(Text)`
  font-weight: bold;
`;

export { Text, TextBold };
