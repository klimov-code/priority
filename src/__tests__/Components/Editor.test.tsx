import React from 'react';
import { render, fireEvent, waitForElement } from 'react-testing-library';
import { Editor } from '~/Components';

describe('Editor', () => {
  it('renders correctly', () => {
    const { container } = render(<Editor />);

    expect(container).toMatchSnapshot();
  });

  it('sets "Name"', () => {
    const { getByPlaceholderText } = render(<Editor name={'first'} />);
    const input = getByPlaceholderText('Name');
    const newName = 'second';

    fireEvent.change(input, { target: { value: newName } });

    expect(input).toHaveProperty('value', newName);
  });

  it('sets "Type"', async () => {
    const { container, getByText } = render(<Editor />);
    const targetType = 'Date range';

    let input: HTMLElement;
    await waitForElement(() => (input = container.querySelector('.type__input input')));
    fireEvent.change(input, { target: { value: 'd' } });

    await waitForElement(() => container.querySelector('.type__menu'));
    const option = getByText(targetType);
    fireEvent.click(option);

    let singleValue: HTMLElement;
    await waitForElement(() => (singleValue = container.querySelector('.type__single-value')));
    expect(singleValue.textContent).toBe(targetType);
  });
});
