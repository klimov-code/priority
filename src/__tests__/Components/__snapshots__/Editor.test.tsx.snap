// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor renders correctly 1`] = `
<div>
  <div
    class="sc-htoDjs hXvUHb"
  >
    <div
      class="sc-dnqmqq jASDld"
    >
      <input
        class="sc-gzVnrw jaSAYM"
        placeholder="Name"
        value=""
      />
      <div
        class="css-1pcexqc-container"
      >
        <div
          class="css-bg1rzq-control type__control"
        >
          <div
            class="css-1hwfws3 type__value-container"
          >
            <div
              class="css-151xaom-placeholder type__placeholder"
            >
              Select priority type
            </div>
            <div
              class="css-1g6gooi"
            >
              <div
                class="type__input"
                style="display: inline-block;"
              >
                <input
                  aria-autocomplete="list"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  id="react-select-2-input"
                  spellcheck="false"
                  style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <div
                  style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                />
              </div>
            </div>
          </div>
          <div
            class="css-1wy0on6 type__indicators"
          >
            <span
              class="css-bgvzuu-indicatorSeparator type__indicator-separator"
            />
            <div
              aria-hidden="true"
              class="css-16pqwjk-indicatorContainer type__indicator type__dropdown-indicator"
            >
              <svg
                aria-hidden="true"
                class="css-19bqh2r"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        style="width: 255px;"
      />
      <div
        class="sc-ifAKCX hOJEll"
        tabindex="0"
      >
        <div
          class="sc-EHOje cpQrJT"
          style="background: rgb(74, 74, 74);"
        />
      </div>
    </div>
    <div
      class="sc-iwsKbI fiAXgu"
    >
      <button
        class="sc-bZQynM hpgkFj"
      >
        Create
      </button>
      <button
        class="sc-bZQynM hpgkFj"
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;
